snuba/request/__init__.py

```diff
@@ -14,6 +14,7 @@ class Request:
     id: str
     body: Mapping[str, Any]
     query: Union[Query, CompositeQuery[Entity]]
+    snql_anonymized: str
```

Comment:
Why do we need this in the request? 
If you fix https://github.com/getsentry/snuba/pull/2282/files#r769854871 this change should go as well, right ?

Reply:
That fix was referring only to avoiding creating a deep copy, the comment below addresses why it's in the Request.